File: .backup/scripts/python/20260309_014820/rollback-ts-migration.py
#!/usr/bin/env python3
"""
rollback-ts-migration.py — Revert TypeScript migration changes.

Usage:
    python3 .backup/scripts/python/20260309_014820/rollback-ts-migration.py [--dry-run]
"""

import argparse
import os
import shutil
import sys
from pathlib import Path


def log(msg: str) -> None:
    print(f"[rollback] {msg}")


def find_workspace() -> Path:
    """Walk up from this script to find the workspace root (has _inc/ dir)."""
    p = Path(__file__).resolve().parent
    for _ in range(10):
        if (p / "_inc").is_dir():
            return p
        p = p.parent
    raise RuntimeError("Cannot find workspace root (no _inc/ directory found)")


def main() -> None:
    parser = argparse.ArgumentParser(description="Rollback TS migration")
    parser.add_argument("--dry-run", action="store_true", help="Show what would be done")
    args = parser.parse_args()

    workspace = find_workspace()
    ts_dir = workspace / "_inc" / "laravel" / "ts"

    log(f"Workspace: {workspace}")
    log(f"TS dir:    {ts_dir}")
    if args.dry_run:
        log("** DRY RUN — no changes will be made **")

    removed = 0

    # 1. Remove dist/ and dist-iife/
    for d in ["dist", "dist-iife"]:
        target = ts_dir / d
        if target.is_dir():
            log(f"Removing {d}/...")
            if not args.dry_run:
                shutil.rmtree(target)
            removed += 1

    # 2. Remove lang TS files
    routes_dir = ts_dir / "src" / "public" / "assets" / "js" / "routes"
    if routes_dir.is_dir():
        lang_files = list(routes_dir.rglob("lang/*.ts"))
        log(f"Found {len(lang_files)} lang TS files")
        if not args.dry_run:
            for f in lang_files:
                f.unlink()
            # Remove empty lang/ dirs
            for d in routes_dir.rglob("lang"):
                if d.is_dir() and not any(d.iterdir()):
                    d.rmdir()
        removed += len(lang_files)

    # 3. Remove core singletons
    core_dir = ts_dir / "src" / "public" / "assets" / "js" / "core"
    if core_dir.is_dir():
        log("Removing core TS singletons...")
        if not args.dry_run:
            shutil.rmtree(core_dir)
        removed += 1

    # 4. Remove integration tests
    integ_dir = ts_dir / "tests" / "integration"
    if integ_dir.is_dir():
        log("Removing integration tests...")
        if not args.dry_run:
            shutil.rmtree(integ_dir)
        removed += 1

    # 5. Remove ESM→IIFE script
    esm_script = ts_dir / "scripts" / "esm-to-iife.cjs"
    if esm_script.is_file():
        log("Removing ESM→IIFE script...")
        if not args.dry_run:
            esm_script.unlink()
        removed += 1

    log("")
    log(f"Rollback complete. {removed} items removed.")
    log("Original JS files in public/assets/js/routes/ are untouched.")


if __name__ == "__main__":
    main()

File: .backup/scripts/php/20260309_014820/rollback-ts-migration.php
#!/usr/bin/env php
<?php
/**
 * rollback-ts-migration.php — Revert TypeScript migration changes.
 *
 * Usage:
 *   php .backup/scripts/php/20260309_014820/rollback-ts-migration.php [--dry-run]
 */

$dryRun = in_array('--dry-run', $argv);

function logMsg(string $msg): void {
    echo "[rollback] {$msg}\n";
}

function findWorkspace(): string {
    $dir = __DIR__;
    for ($i = 0; $i < 10; $i++) {
        if (is_dir($dir . '/_inc')) {
            return $dir;
        }
        $dir = dirname($dir);
    }
    throw new RuntimeException('Cannot find workspace root');
}

function rmDirRecursive(string $dir): void {
    if (!is_dir($dir)) return;
    $items = new RecursiveIteratorIterator(
        new RecursiveDirectoryIterator($dir, RecursiveDirectoryIterator::SKIP_DOTS),
        RecursiveIteratorIterator::CHILD_FIRST
    );
    foreach ($items as $item) {
        if ($item->isDir()) {
            rmdir($item->getPathname());
        } else {
            unlink($item->getPathname());
        }
    }
    rmdir($dir);
}

function findLangFiles(string $dir): array {
    $result = [];
    if (!is_dir($dir)) return $result;
    $iterator = new RecursiveIteratorIterator(
        new RecursiveDirectoryIterator($dir, RecursiveDirectoryIterator::SKIP_DOTS)
    );
    foreach ($iterator as $file) {
        $path = $file->getPathname();
        if (preg_match('#/lang/[^/]+\.ts$#', $path)) {
            $result[] = $path;
        }
    }
    return $result;
}

// Main
$workspace = findWorkspace();
$tsDir = $workspace . '/_inc/laravel/ts';

logMsg("Workspace: {$workspace}");
logMsg("TS dir:    {$tsDir}");
if ($dryRun) logMsg("** DRY RUN — no changes will be made **");

$removed = 0;

// 1. dist/ and dist-iife/
foreach (['dist', 'dist-iife'] as $d) {
    $target = "{$tsDir}/{$d}";
    if (is_dir($target)) {
        logMsg("Removing {$d}/...");
        if (!$dryRun) rmDirRecursive($target);
        $removed++;
    }
}

// 2. Lang TS files
$routesDir = "{$tsDir}/src/public/assets/js/routes";
$langFiles = findLangFiles($routesDir);
logMsg("Found " . count($langFiles) . " lang TS files");
if (!$dryRun) {
    foreach ($langFiles as $f) {
        unlink($f);
    }
}
$removed += count($langFiles);

// 3. Core singletons
$coreDir = "{$tsDir}/src/public/assets/js/core";
if (is_dir($coreDir)) {
    logMsg("Removing core TS singletons...");
    if (!$dryRun) rmDirRecursive($coreDir);
    $removed++;
}

// 4. Integration tests
$integDir = "{$tsDir}/tests/integration";
if (is_dir($integDir)) {
    logMsg("Removing integration tests...");
    if (!$dryRun) rmDirRecursive($integDir);
    $removed++;
}

// 5. ESM→IIFE script
$esmScript = "{$tsDir}/scripts/esm-to-iife.cjs";
if (is_file($esmScript)) {
    logMsg("Removing ESM→IIFE script...");
    if (!$dryRun) unlink($esmScript);
    $removed++;
}

logMsg("");
logMsg("Rollback complete. {$removed} items removed.");
logMsg("Original JS files in public/assets/js/routes/ are untouched.");
